packages/dd-trace/src/datastreams/pathway.js

```diff
@@ -0,0 +1,44 @@
+// encoding used here is FNV1a
+// other languages use FNV1
+// this inconsistency is ok because hashes do not need to be consistent across services
+const fnv = require('fnv-plus')
+const { encodeVarint, decodeVarint } = require('./encoding')
+
+function fnvHash (checkpointString) {
+  const hash = fnv.hash(checkpointString, 64)
+  return Buffer.from(hash.hex(), 'hex')
+}
+
+function computeHash (service, env, edgeTags, parentHash) {
```

Comment:
Could this be cached? Otherwise what if you are sending thousands of messages to Kafka? Should we also add a benchmark to see how much overhead this is adding?

Reply:
We can use lru-cache to cache these

```js
function computeHash (service, env, edgeTags, parentHash) {
  const key = `${service}${env}` + edgeTags.join('') + parentHash.toString

  if (cache.has(key)) {
    return cache.get(key)
  }
  const currentHash = fnvHash(`${service}${env}` + edgeTags.join(''))
  const buf = Buffer.concat([ currentHash, parentHash ], 16)
  const val = fnvHash(buf.toString())
  cache.set(key, val)
  return val
}
```
```